src/app/copy/copy-base.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  EventEmitter,
+  Input,
+  Output,
+} from '@angular/core';
+
+import { CopyService } from './copy-service/copy.service';
+
+export abstract class CopyBase {
+  @Input('copyBtnAriaLabel') copyBtnAriaLabel: string;
+  @Input('copyValue') copyValue: string = 'Missing \'copyValue\' @Input property';
+  @Input('tooltip') tooltip: string;
+  @Input('tooltipPlacement') tooltipPlacement: string = 'top';
+
+  /**
+   * Event emitted with the chart reference after load is complete
+   * @type {EventEmitter}
+   */
+  @Output('copiedToClipboard') copiedToClipboard: EventEmitter<any> = new EventEmitter();
+
+  private _recentlyCopied: boolean = false;
+
+  /**
+   * Returns the flag indicating copy action has just happened
+   *
+   * @returns {boolean} True if copy action has been triggered
+   */
+  public get recentlyCopied(): boolean {
+    return this._recentlyCopied;
+  }
+
+  /**
+   * Set the flag indicating copy action has just happened
+   *
+   * @param {boolean} copied True when copy action has been triggered
+   */
+  public set recentlyCopied(copied: boolean) {
+    this._recentlyCopied = copied;
+  }
+
+  /**
+   * Default constructor
+   */
+  constructor(protected copyService: CopyService) {}
+
+  /**
+   * Copy value to the user's system clipboard
+   */
+  copyValueToClipboard(accessibleName: string): void {
```

Comment:
Can this method be marked protected?

Reply:
So when I use protected, I get the aot error

> ERROR in ng:///workingrepos/seanforyou23-patternfly-ng/src/app/copy/inline-copy/inline-copy.component.html (13,5): Property 'copyValueToClipboard' is protected and only accessible within class 'InlineCopyComponent' and its subclasses.

The default seemed to work ok so I was going with that. I'm looking into it more now though, feel free to make a suggestion if this is a familiar error

Reply:
Instead of calling copyValueToClipboard directly within the HTML template, you could do something more like this:

copyValue(): void {
  this.copyValueToClipboard(this.copyBtnAriaLabel);
}

Reply:
Ah ok, I see what you mean. I was actually considering doing this before but with a new build I'm realizing that error was just a one-off. Thanks for explaining that alternative though.

Reply:
Oh, I see in your other comment there is preference to make these protected, so yea I'll be taking this advice after all.